nixos/lib/build-vms.nix

```diff
@@ -28,7 +37,8 @@ rec {
           ../modules/testing/test-instrumentation.nix # !!! should only get added for automated test runs
           { key = "no-manual"; documentation.nixos.enable = false; }
           { key = "qemu"; system.build.qemu = qemu; }
-        ] ++ optional minimal ../modules/testing/minimal-kernel.nix;
+        ] ++ optional minimal ../modules/testing/minimal-kernel.nix
+          ++ extraConfigurations;
```

Comment:
This shouldn't be necessary. Just add on to configurations above.

Reply:
In order to do so, we seem to have two options to choose from:

a. We have to change the definition of a VM test, such that a machine can be specified not only as a single module, but as a list of modules. This is unusual because it doesn't seem to work that way in other tools.
b. We wrap the user's machine in a module that imports both the user's machine and the extra configuration. This means we have to accept that the tree structure of imports gets more complicated, impacting the error messages about the modules of which the name can not be determined.

And then there's the third option, leaving the PR as is wrt `extraConfigurations` with the only downside being two extra lines of code. Also, we'll have a clear, documented method of setting default configuration in a test runner.

Yet another idea is to add a `defaults` configuration like in NixOps, to be defined by the test author, but, to be honest, I'm not sure whether that's actually a good feature. Also, it only *shifts* the above problem to a single module.

Reply:
I'm just saying to get rid of extraConfigurations in this file but define it testing.nix by appending to the "configurations" list here:

https://github.com/NixOS/nixpkgs/pull/47684/files#diff-4f17a8db1c53e375c043d58e86ec205cR172

Reply:
That doesn't solve it because `nixosTest` needs `makeTest` which calls `buildVirtualNetwork`, not the `runInMachine` function that you linked.